### PR TITLE
OSD-20029 refactor and add MacOS monitoring plugin support

### PR DESCRIPTION
**Pull request by feichashao**
### What type of PR is this?

cleanup

### What this PR does / Why we need it?

Refactor the code into different functions and container engine implementations.

This also fixes the bug in [OSD-20029](https://issues.redhat.com//browse/OSD-20029).

The current code assumes little difference between podman vs docker and Linux vs MacOS, but with the growth of the logic, we found the difference is large and the all-in-one function logic is hard to understand.

With the refactor, 
- The mainly logic of cluster related handling lands in `run()`.
- The container-engine-specific logic are implemented using `containerEngineInterface`, each Container Engine + OS combination has its own implementation.

Specifically,
- For docker, we need to run in the host network, and the authfile directory should be specified by `--config` in front of the sub-command.
- For podman in MacOS, as it is running in a virtual machine, we need to put the nginx file to the virtual machine for container to consume.

### Which Jira/Github issue(s) does this PR fix?

[OSD-20029](https://issues.redhat.com//browse/OSD-20029)

### Pre-checks (if applicable)

- [x] Ran unit tests locally
- [x] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

File: cmd/ocm-backplane/console/console.go [hunk 1]
```diff
@@ -924,41 +857,424 @@ func isRunningHigherOrEqualTo(targetVersionStr string) bool {
...
+	if err != nil {
+		return err
+	}
+	nginxConfPath := filepath.Join(configDirectory, nginxConf)
+	return dockerRunMonitorPlugin(containerName, consoleContainerName, nginxConfPath, pluginArgs)
+}
+
+// put a file in place for container to mount
+// filename should be name only, not a path
+func (ce *podmanLinux) putFileToMount(filename string, content []byte) error {
```
[the remark this answers is not in the document]
Comment (feichashao) [reply cmd/ocm-backplane/console/console.go:1174]: Added debug log to podman on Linux and docker in the new commit.

File: cmd/ocm-backplane/console/console.go [hunk 2]
```diff
@@ -44,386 +48,437 @@ import (
...
-	runCmd := createCommand(containerEngine, engRunArgs...)
-	runCmd.Stderr = os.Stderr
-	runCmd.Stdout = nil
-
-	err = runCmd.Run()
-	if err != nil {
-		return err
+func (o *consoleOptions) determineMonitorPluginImage(config *rest.Config) error {
+	// we don't need this for lower than 4.14
+	if !isRunningHigherOrEqualTo(versionForMonitoringPlugin) {
```
[the remark this answers is not in the document]
Comment (feichashao) [reply cmd/ocm-backplane/console/console.go]: Moved it to `o.determineNeedMonitorPlugin()`

**Comment by samanthajayasinghe** [comment]
@feichashao Thanks for fixing the console for Mac and Linux .
As an improvement ( not in this PR ), We could move relevant functions/struct to console-mac.go and console-linux.go files.

**Pull request merged**